artifactory/commands/transferinstall/installplugin.go

```diff
@@ -0,0 +1,369 @@
+	"github.com/jfrog/jfrog-cli-core/v2/utils/config"
+	"github.com/jfrog/jfrog-cli-core/v2/utils/coreutils"
+	"github.com/jfrog/jfrog-client-go/artifactory"
+	"github.com/jfrog/jfrog-client-go/utils/errorutils"
+	"github.com/jfrog/jfrog-client-go/utils/io/fileutils"
+	"github.com/jfrog/jfrog-client-go/utils/log"
+	"github.com/pkg/errors"
+	"net/http"
+	url "net/url"
+	"os"
+	"path"
+	"path/filepath"
+	"strings"
+)
+
+const (
+	minArtifactoryVersion = "2.9.0" // for reload api (version api is from 2.2.2)
+	pluginReloadRestApi   = "api/plugins/reload"
+	jfrogHomeEnvVar       = "JFROG_HOME"
+	latest                = "[RELEASE]"
+)
+
+var (
+	defaultSearchPath = filepath.Join("opt", "jfrog")
+	// Plugin directory locations
+	originalDirPath = PluginFileItem{"artifactory", "etc", "plugins"}
+	v7DirPath       = PluginFileItem{"artifactory", "var", "etc", "artifactory", "plugins"}
+	// Error types
+	emptyUrlErr            = errors.Errorf("Base download URL must be provided to allow file downloads.")
+	notValidDestinationErr = func(showHint bool) error {
+		hint := ""
+		if showHint {
+			hint = "Hint: use --home-dir option."
+		}
+		return errors.Errorf("Can't find target plugin directory, this command must run on a machine with Artifactory server. %s", hint)
+	}
+	minVerErr             = errorutils.CheckErrorf("This operation requires Artifactory version %s or higher", minArtifactoryVersion)
+	downloadConnectionErr = func(baseUrl string) error {
+		return errorutils.CheckErrorf("We tried to download the plugin files from '%s' but got connection issue. Hint: manual transfer the files to the machine and use --source-dir option.", baseUrl)
+	}
+)
+
+type PluginFileItem []string
+type PluginFiles []PluginFileItem
+
+// Get the name componenet of the item
+func (f *PluginFileItem) Name() string {
+	size := len(*f)
+	if size == 0 {
+		return ""
+	}
+	return (*f)[size-1]
+}
+
+// Get the directory list of the item, ignore empty entries
+func (f *PluginFileItem) Dirs() *PluginFileItem {
+	dirs := PluginFileItem{}
+	for i := 0; i < len(*f)-1; i++ {
+		dir := (*f)[i]
+		if dir != "" {
+			dirs = append(dirs, dir)
+		}
+	}
+	return &dirs
+}
+
+// Split and get the componenets of the item
+func (f *PluginFileItem) SplitNameAndDirs() (string, *PluginFileItem) {
+	return f.Name(), f.Dirs()
+}
+
+// Convert the item to URL representation, ignore empty entries, adding prefix tokens as provided
+func (f *PluginFileItem) toURL(prefixUrl string) (string, error) {
+	myUrl, err := url.Parse(prefixUrl)
+	if err != nil {
+		return "", err
+	}
+	myUrl.Path = path.Join(myUrl.Path, path.Join(*f...))
+	return myUrl.String(), nil
+}
+
+// Convert the item to path representation, ignore empty entries, adding prefix tokens as provided
+func (f *PluginFileItem) toPath(previousTokens ...string) string {
+	return filepath.Join(filepath.Join(previousTokens...), filepath.Join(*f.Dirs()...), f.Name())
+}
+
+// Holds all the plugin files and options of destinations to transfer them into
```

Comment:
"and options of destinations to transfer them into" - this is really unclear. Let's try to elaborate more on what this means. Maybe add some examples.

Reply:
done